pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1079,6 +1091,18 @@ private void deleteBundleDataFromMetadataStore(String bundle) {
         }
     }
 
+    private void deleteTimeAverageDataFromMetadataStore(String broker) {
+        final String timeAverageZPath = TIME_AVERAGE_BROKER_ZPATH + "/" + broker;
+        try {
+            timeAverageBrokerDataCache.delete(timeAverageZPath).join();
```

Comment:
The `updateAllBrokerData` will be called in the metadata listener thread, looks like we using .join() here might introduce a deadlock.

Reply:
done, PTAL @codelipenghui